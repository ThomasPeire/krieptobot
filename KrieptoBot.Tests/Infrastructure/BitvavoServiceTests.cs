using System;
using System.Threading.Tasks;
using FluentAssertions;
using KrieptoBot.Infrastructure.Bitvavo.Services;
using KrieptoBot.Tests.Mocks.Bitvavo;
using NUnit.Framework;
using Snapshooter.NUnit;

namespace KrieptoBot.Tests.Infrastructure
{
    public class BitvavoServiceTests
    {
        private readonly MockBitvavoApi _mockService = new();
        private BitvavoService _bitvavoService;

        [SetUp]
        public void Setup()
        {
            _mockService.InitData();
            _bitvavoService = new BitvavoService(_mockService);
        }

        [Test]
        public async Task GetBalance_ShouldReturn_Balance()
        {
            var result = await _bitvavoService.GetBalanceAsync();

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetCandles_ShouldReturn_Candles()
        {
            var result = await _bitvavoService.GetCandlesAsync("BTC-EUR");

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetMarkets_ShouldReturn_Markets()
        {
            var result = await _bitvavoService.GetMarketsAsync();

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetMarket_ShouldReturn_Market()
        {
            var result = await _bitvavoService.GetMarketAsync("BTC-EUR");

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetAssets_ShouldReturn_Assets()
        {
            var result = await _bitvavoService.GetAssetsAsync();

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetAsset_ShouldReturn_Asset()
        {
            var result = await _bitvavoService.GetAssetAsync("BTC");

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetTrades_ShouldReturn_Trades()
        {
            var result = await _bitvavoService.GetTradesAsync("BTC-EUR");

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetOrders_ShouldReturn_Orders()
        {
            var result = await _bitvavoService.GetOrdersAsync("BTC-EUR");

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetOrder_ShouldReturn_Order()
        {
            var result =
                await _bitvavoService.GetOrderAsync("BTC-EUR", new Guid("4a7bd126-2d21-4918-96dc-0c8f51760a0b"));

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetOpenOrder_ShouldReturn_OpenOrder()
        {
            var result = await _bitvavoService.GetOpenOrderAsync("BTC-EUR");

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task PostOrder_ShouldReturn_AddOrder()
        {
            var order = await _bitvavoService.PostBuyOrderAsync("BTC-EUR", "Limit", 9000.00m, 0.0001m);

            var result = await _bitvavoService.GetOrderAsync(order.MarketName.Value, order.Id);

            Assert.That(order.MarketName.Value, Is.EqualTo(result.MarketName.Value));
            Assert.That(order.Side, Is.EqualTo(result.Side));
            Assert.That(order.Type, Is.EqualTo(result.Type));
            Assert.That(order.Amount, Is.EqualTo(result.Amount));
            Assert.That(order.Price, Is.EqualTo(result.Price));
        }
    }
}