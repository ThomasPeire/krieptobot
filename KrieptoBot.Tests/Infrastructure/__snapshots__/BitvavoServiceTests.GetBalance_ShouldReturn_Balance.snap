[
  {
    "Symbol": "EUR",
    "Available": 0.0,
    "InOrder": 11.03
  },
  {
    "Symbol": "ADA",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "ARK",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "BAT",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "BTC",
    "Available": 0.00000024,
    "InOrder": 0.0
  },
  {
    "Symbol": "CHZ",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "CMT",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "ETH",
    "Available": 0.01959154,
    "InOrder": 0.0
  },
  {
    "Symbol": "GAS",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "HOT",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "LTC",
    "Available": 0.00011392,
    "InOrder": 0.0
  },
  {
    "Symbol": "NPXS",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "TRX",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "XEM",
    "Available": 0.0,
    "InOrder": 0.0
  }
]
